indico/queries/model_groups/model_groups.py

```diff
@@ -236,3 +236,50 @@ def requests(self):
         yield _ModelGroupPredict(
             model_id=self.model_id, data=self.data, predict_options=self.predict_options
         )
+
+
+class UpdateModelGroupSettings(GraphQLRequest):
+    """
+    Updates an existing model group component in the platform.
+
+    Args:
+        model_group_id (int): the id of the model group to update settings
+        model_training_options (dict): model training options to use when training model
+    """
+
+    query = """
+        mutation updateModelGroup($modelGroupId: Int!, $modelTrainingOptions: JSONString) {
+            updateModelGroupSettings(
+                modelGroupId: $modelGroupId
+                modelTrainingOptions: $modelTrainingOptions
+            ) {
+                modelOptions {
+                    id
+                    modelTrainingOptions
+                }
+            }
+        }
+    """
+
+    def __init__(
+        self,
+        model_group_id: int,
+        model_training_options: dict = None,
```

Comment:
typing for a default None should always include None in the option 
```suggestion
        model_training_options: dict | None = None,
```
also can't remember if the sdk is supposed to remain 3.8 compatible in which case you'll need to use the typing library (Optional[Dict[str, Any]]) instead